qiskit/dagcircuit/dagcircuit.py

```diff
@@ -786,17 +786,30 @@ def idle_wires(self, ignore=None):
 
         Yields:
             Bit: Bit in idle wire.
+
+        Raises:
+            DAGCircuitError: If the DAG is invalid
         """
         if ignore is None:
-            ignore = []
+            ignore = set()
+        ignore_set = set(ignore)
         for wire in self._wires:
-            nodes = [
-                node
-                for node in self.nodes_on_wire(wire, only_ops=True)
-                if node.op.name not in ignore
-            ]
-            if len(nodes) == 0:
-                yield wire
+            if not ignore:
+                try:
+                    child = next(self.successors(self.input_map[wire]))
+                except StopIteration as e:
+                    raise DAGCircuitError(
+                        "Invalid dagcircuit input node %s has no output" % self.input_map[wire]
+                    ) from e
```

Comment:
Out of curiosity, we don't usually guard against a malformed DAGCircuit, was there a reason to here?

Reply:
Pylint was yelling at me locally about raising stopiteration without it. This made it happy...